### PR TITLE
Remove Combine from System command running implementation.

### PR DESCRIPTION
**Pull request by waltflanagan**
### Short description 📝

As discussed [here](https://tuist-community.slack.com/archives/CTB0DBVD2/p1716644981994629), this PR removes Combine and CombineExt from our System implementation for running commands.  This was motivated by our move towards strict concurrency checking and to align on using async/await for our concurrent programming approach.  

### How to test the changes locally 🧐

This should not have any impact on behaviors within Tuist.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistSupport/System/System.swift [hunk 1]
```diff
@@ -260,19 +210,26 @@ public final class System: Systeming {
...
-        verbose: Bool,
-        environment: [String: String],
-        redirection: TSCBasic.Process.OutputRedirection
+        verbose: Bool = false,
+        environment: [String: String] = ProcessInfo.processInfo.environment,
+        redirection: TSCBasic.Process.OutputRedirection = .none
     ) throws {
+        // We have to collect both stderr and stdout because we want to stream the output
+        // the `ProcessResult` type will not contain either unless outputRedirection is set to `.collect`
+        var errorData: [UInt8] = []
```
Comment (pepicrft) [inline Sources/TuistSupport/System/System.swift:221]:
```suggestion
        var stdErrData: [UInt8] = []
```

File: Sources/TuistSupport/System/System.swift [hunk 2]
```diff
@@ -260,19 +210,26 @@ public final class System: Systeming {
...
+        var stdOutData: [UInt8] = []
+
         let process = Process(
             arguments: arguments,
             environment: environment,
             outputRedirection: .stream(stdout: { bytes in
+                stdOutData.append(contentsOf: bytes)
```
Comment (pepicrft) [inline Sources/TuistSupport/System/System.swift:228]: One potential issue with doing this by default, is that we might overflow the memory with processes like `xcodebuild`'s, which outputs a lot of data. What if we make this controllable by the caller?

File: Sources/TuistSupportTesting/Utils/MockSystem.swift [hunk 3]
```diff
@@ -1,4 +1,3 @@
-import Combine
```
Comment (pepicrft) [inline Sources/TuistSupportTesting/Utils/MockSystem.swift:1]: What do you think if we get rid of this, and instead use the `Mockable` macro?

**Review by pepicrft — APPROVED** [review]
This is amazing. Thanks a lot for doing this @waltflanagan

File: Sources/TuistSupport/System/System.swift (hunk 2 above)
Comment (waltflanagan) [reply Sources/TuistSupport/System/System.swift:228]: For now i can just remove it, i added it for completeness.  I'll take it into account with the refactor i mentioned in Slack.

**Review by danieleformichelli — APPROVED** [review]

**Pull request merged**